HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -6755,6 +6837,40 @@ def initialize(hpxml_element, *args, **kwargs)
     attr_reader(*CLASS_ATTRS)
     attr_accessor(*ATTRS)
 
+    # Returns any branch circuits that the component may be attached to.
+    #
+    # @return [Array<HPXML::BranchCircuit>] List of branch circuit objects
+    def branch_circuits
+      list = []
+      @parent_object.electric_panels.each do |electric_panel|
+        electric_panel.branch_circuits.each do |branch_circuit|
+          next if branch_circuit.component_idrefs.nil? || branch_circuit.component_idrefs.empty?
+          next unless branch_circuit.component_idrefs.include?(@id)
+
+          list << branch_circuit
+        end
+      end
+
+      return list
+    end
+
+    # Returns any service feeders that the component may be attached to.
+    #
+    # @return [Array<HPXML::ServiceFeeder>] List of service feeder objects
+    def service_feeders
+      list = []
+      @parent_object.electric_panels.each do |electric_panel|
+        electric_panel.service_feeders.each do |service_feeder|
+          next if service_feeder.component_idrefs.empty?
+          next unless service_feeder.component_idrefs.include?(@id)
+
+          list << service_feeder
+        end
+      end
+
+      return list
+    end
```

Comment:
The same logic repeated over and over and over and over? Please use a method.

Reply:
https://github.com/NREL/OpenStudio-HPXML/pull/1946/commits/33c0888896129988062d57ab16ac4e08652af37a